Add a DELAY_KEY option to the spec

kelvins opened the pull request:
- Add a `delay` key option to the `endpoints` level to perform a delay to each request in that endpoint.
- The idea is to set the `delay` key (yaml spec) in milliseconds but convert it to seconds to use it in the `time.sleep`.
- I'm not sure if it is better to convert the `delay` in the property (when getting it) or in the `time.sleep` call (when using it). Any suggestions are welcome. 
- Close https://github.com/scanapi/scanapi/issues/266.

camilamaia commented:
@kelvins super good PR, thanks for it. This is a feature that a bunch of people is asking for. The way you implemented it, converting it in the property, works fine for me 👍 

I would only add the possibility of having the delay key in the `requests` level too. What do you think? So if a request has a delay key, it would override the endpoints one. The same way you implemented for endpoint parent and child: https://github.com/scanapi/scanapi/pull/292/files#diff-ea44743607e97fd3f62c98cb9c9d5ad7R83

kelvins commented:
@camilamaia thanks for the review, I added the delay key to both `RequestNode` and `EndpointNode` as you suggested. Since we now have two properties to get the delay, I moved the seconds to milliseconds conversion to the `time.sleep` call, to avoid doing this twice (for each property).

camilamaia reviewed (approved):
Really awesome pr, thanks @kelvins! I will release a new version soon since this feature is quite requested!
